audiomentations/augmentations/room_simulator.py

```diff
@@ -0,0 +1,355 @@
+            0, min(self.parameters["size_y"] - self.padding, mic_y)
+        )
+        self.parameters["mic_z"] = max(
+            0, min(self.parameters["size_z"] - self.padding, mic_z)
+        )
+
+    def apply(self, samples, sample_rate):
+        try:
+            import pyroomacoustics as pra
+        except ImportError:
+
+            print(
+                "Failed to import pyroomacoustics. Maybe it is not installed? "
+                "To install the optional pyroomacoustics dependency of audiomentations,"
+                " do `pip install audiomentations[extras]` or simply "
+                " `pip install pyroomacoustics`",
+                file=sys.stderr,
+            )
+            raise
+
+        assert samples.dtype == np.float32
+
+        # It makes no sense to use stereo data.
+        if len(samples.shape) > 1:
+            samples = samples.mean(0)
+
+        # Construct room
+        self.room = pra.Room.from_corners(
+            np.array(
+                [
+                    [0, 0],
+                    [0, self.parameters["size_x"]],
+                    [self.parameters["size_x"], self.parameters["size_y"]],
+                    [self.parameters["size_y"], 0],
+                ]
+            ).T,
+            fs=sample_rate,
+            materials=pra.Material(self.parameters["absorption_coefficient"]),
+            ray_tracing=self.use_ray_tracing,
+            air_absorption=True,
+        )
+
+        if self.use_ray_tracing:
+            # TODO: Somehow make those parameters
+            self.room.set_ray_tracing(
+                receiver_radius=0.5, n_rays=10000, energy_thres=1e-5
+            )
+
+        self.room.extrude(
+            height=self.parameters["size_z"],
+            materials=pra.Material(self.parameters["absorption_coefficient"]),
+        )
+
+        # Add the point source
+        self.room.add_source(
+            np.array(
+                [
+                    self.parameters["source_x"],
+                    self.parameters["source_y"],
+                    self.parameters["source_z"],
+                ]
+            ),
+            signal=samples,
+        )
+
+        # Add the microphone
+        self.room.add_microphone_array(
+            pra.MicrophoneArray(
+                np.array(
+                    [
+                        [
+                            self.parameters["mic_x"],
+                            self.parameters["mic_y"],
+                            self.parameters["mic_z"],
+                        ]
+                    ]
+                ).T,
+                self.room.fs,
+            )
+        )
+
+        # Do the simulation
+        self.room.compute_rir()
+        self.room.simulate()
+        result = self.room.mic_array.signals.astype(np.float32).flatten()
+
+        num_samples = max(samples.shape)
+
+        if self.leave_length_unchanged:
+            return result[:num_samples]
+        return result
```

Comment:
What if the input was a 2D array? Will it be a 1D array here, since you did `samples = samples.mean(0)` earlier?

Reply:
Yes. However I will try changing it as in ApplyImpulseResponse like you suggested above